source/Engine.cpp

```diff
@@ -1236,6 +1233,35 @@ void Engine::EnterSystem()
 	player.IncrementDate();
 	const Date &today = player.GetDate();
 
+	if(player.HasTravelPlan() && player.GetSystem() == player.TravelPlan().back())
+		player.PopTravel();
```

Comment:
This was done in Step to prevent a race condition, and I'm pretty sure one can occur in certain circumstances if you do this here.